Test-Source/Invoke-DockerPull.Tests.ps1

```diff
@@ -27,115 +19,90 @@ Describe 'Pull docker images' {
-            }
+            $theCode = { Invoke-DockerPull -ImageName 'ubuntu' -Tag 'bionic' -Digest 'sha256:f5c0a8d225a4b7556db2b26753a7f4c4de3b090c1a8852983885b80694ca9840' }
             $theCode | Should -Throw -ExceptionType ([System.Management.Automation.ParameterBindingException]) -PassThru
         }
 
         It 'pulls public docker image by image name with invalid digest, missing sha256: prefix; and fails' {
-            $theCode = {
-                Invoke-DockerPull -ImageName 'ubuntu' -Digest 'f5c0a8d225a4b7556db2b26753a7f4c4de3b090c1a8852983885b80694ca9840'
-            }
+            $theCode = { Invoke-DockerPull -ImageName 'ubuntu' -Digest 'f5c0a8d225a4b7556db2b26753a7f4c4de3b090c1a8852983885b80694ca9840' }
             $theCode | Should -Throw -ExceptionType ([System.Management.Automation.RuntimeException]) -PassThru
         }
 
         It 'pulls public docker image by image name with invalid digest, wrong digest length; and fails' {
-            $theCode = {
-                Invoke-DockerPull -ImageName 'ubuntu' -Digest 'sha256:f5c0a8d225a4b7556db2b26753a7f4c4d'
-            }
+            $theCode = { Invoke-DockerPull -ImageName 'ubuntu' -Digest 'sha256:f5c0a8d225a4b7556db2b26753a7f4c4d' }
             $theCode | Should -Throw -ExceptionType ([System.Management.Automation.RuntimeException]) -PassThru
         }
 
         It 'does not allow colons in imagename, force use of tag' {
-            $theCode = {
-                Invoke-DockerPull -ImageName 'ubuntu:bionic'
-            }
+            $theCode = { Invoke-DockerPull -ImageName 'ubuntu:bionic' }
             $theCode | Should -Throw -ExceptionType ([System.ArgumentException]) -PassThru
         }
 
         It 'does not allow at signs in imagename, force use of tag' {
-            $theCode = {
-                Invoke-DockerPull -ImageName 'ubuntu@sha256:f5c0a8d225a4b7556db2b26753a7f4c4d'
-            }
+            $theCode = { Invoke-DockerPull -ImageName 'ubuntu@sha256:f5c0a8d225a4b7556db2b26753a7f4c4d' }
             $theCode | Should -Throw -ExceptionType ([System.ArgumentException]) -PassThru
         }
 
         It 'cannot pull the requested docker image, throws exception on non-zero exit code' {
-            $returnExitCodeOne = {
-                $commandResult = [CommandResult]::new()
-                $commandResult.ExitCode = 1
-                return $commandResult
-            }
-            Mock -CommandName "Invoke-Command" $returnExitCodeOne  -Verifiable -ModuleName $Global:ModuleName
-            $theCode = {
-                Invoke-DockerPull -ImageName 'mcr.microsoft.com/ubuntu'
-            }
+            Mock -CommandName "Invoke-Command" $Global:CodeThatReturnsExitCodeOne  -Verifiable -ModuleName $Global:ModuleName
+            $theCode = { Invoke-DockerPull -ImageName 'mcr.microsoft.com/ubuntu' }
```

Comment:
Should we consider refactoring this code to use constants instead of explicit image names?

Reply:
Yes we can. However these image names are made up for the tests. :sweat_smile: